components/table/hooks/useFilter/index.tsx

```diff
@@ -34,7 +34,9 @@ function collectFilterStates<RecordType>(
     } else if (column.filters || 'filterDropdown' in column || 'onFilter' in column) {
       if ('filteredValue' in column) {
         // Controlled
-        const filteredValues = (column.filteredValue || []).map(String);
+        const filteredValues = Array.isArray(column.filteredValue)
+          ? column.filteredValue.map(String)
```

Comment:
@Meowu @afc163  In my case, the custom autocomplete filter drop-down would expect an array of objects ([ {key: 1, label: 'Item 1'}, {key: 2, label: 'Item 2'}]). It resembles  a controlled value of autocomplete component which has labelInValue prop. This code converted value to **["[object Object]", "[object Object]"]** :cry:

Reply:
orz..
I didn't think about this edge case.It's complicated.Would you mind providing a sample code.

Reply:
@Meowu I have reproduced a simple example.
https://codesandbox.io/s/filter-and-sorter-antd4100-forked-wfkx5

Reply:
In my other case, the onChange event of the table isn't triggered when the type of filtered value is object. [ click me](https://prnt.sc/wgiscm)

Reply:
OK.I will do more test and fix it.

Reply:
will fix in https://github.com/ant-design/ant-design/pull/28850 .